pkg/metrics/metrics.go

```diff
@@ -143,6 +144,13 @@ func (m *Metrics) Start(listenAddress string) (*http.Server, error) {
 
 	router := mux.NewRouter()
```

Comment:
can we create a new `NewServeMux` instead of `NewRouter`? so that we can use the `profiling` util.

Reply:
Can you use the profiling util on this `mux`? The one that is used here as `mux.NewRouter()`?

Reply:
profiling util accepts NewServeMux. waiting for @meyskens review on this

Reply:
Not too sure what the historical reason for using the mux libarary was if promhttp just supports the standard libarary one 3rd party dependancy less would be nice

Reply:
Sounds like we agree that we should ditch the gorilla/mux package.
In which case, please go ahead and make that change.